crates/wasmtime/src/compile.rs

```diff
@@ -419,6 +423,26 @@ impl<'a> CompileInputs<'a> {
         ret
     }
 
+    fn clean_symbol(name: &str) -> Cow<str> {
+        // Just to be on the safe side, avoid passing user-provided data to tools
+        // like "perf" or "objdump", and filter the name.  Let only characters usually
+        // used for function names, plus some characters that might be used in name
+        // mangling.
+        let bad_char = |c: char| !c.is_alphanumeric() && !r"<>[]_-:@$".contains(c);
```

Comment:
Maybe you wanted [is_ascii_alphanumeric](https://doc.rust-lang.org/std/primitive.char.html#method.is_ascii_alphanumeric) here instead, rather than allowing all Unicode alpha/numeric codepoints?

```suggestion
        let bad_char = |c: char| !c.is_ascii_alphanumeric() && !r"<>[]_-:@$".contains(c);
```

Reply:
Done!